vsphere/internal/helper/contentlibrary/content_library_helper.go

```diff
@@ -142,21 +151,270 @@ func CreateLibraryItem(c *rest.Client, l *library.Library, name string, desc str
 		Name:        name,
 		Type:        t,
 	}
+	uploadSession := libraryUploadSession{
+		ContentLibraryManager: clm,
+		RestClient:            c,
+		LibraryID:             l.ID,
+	}
+	if moid != "" {
+		return uploadSession.cloneTemplate(moid, name, t)
+	}
+
 	id, err := clm.CreateLibraryItem(ctx, item)
 	if err != nil {
-		return "", provider.ProviderError(name, "CreateLibraryItem", err)
+		return nil, provider.ProviderError(name, "CreateLibraryItem", err)
 	}
 	session, err := clm.CreateLibraryItemUpdateSession(ctx, library.Session{LibraryItemID: id})
 	if err != nil {
-		return "", provider.ProviderError(name, "CreateLibraryItem", err)
+		return nil, provider.ProviderError(name, "CreateLibraryItem", err)
+	}
+	defer func() {
+		_ = clm.CompleteLibraryItemUpdateSession(ctx, session)
+	}()
+	uploadSession.UploadSession = session
+	defer func() {
+		_ = clm.CompleteLibraryItemUpdateSession(ctx, session)
```

Comment:
Why is this called twice?

Reply:
There should only be one. I made a copy/paste error when splitting the PRs. Got it fixed.